src/test/java/pl/gov/coi/pomocua/ads/transport/TransportResourceTest.java

```diff
@@ -0,0 +1,233 @@
+                .build());
+        postOffer(aTransportOffer()
+                .destination(new Location("Pomorskie", "Wejherowo"))
+                .build());
+
+        TransportOfferSearchCriteria searchCriteria = new TransportOfferSearchCriteria();
+        searchCriteria.setDestination(new Location("Pomorskie", "Gdynia"));
+        TransportOffer[] results = searchOffers(searchCriteria);
+
+        assertThat(results.length).isEqualTo(1);
+        assertThat(results[0]).isEqualTo(transportOffer1);
+    }
+
+    @Test
+    void shouldFindByCapacity() {
+        TransportOffer transportOffer1 = postOffer(aTransportOffer()
+                .capacity(10)
+                .build());
+        TransportOffer transportOffer2 = postOffer(aTransportOffer()
+                .capacity(11)
+                .build());
+        postOffer(aTransportOffer()
+                .capacity(1)
+                .build());
+
+        TransportOfferSearchCriteria searchCriteria = new TransportOfferSearchCriteria();
+        searchCriteria.setCapacity(10);
+        TransportOffer[] results = searchOffers(searchCriteria);
+
+        assertThat(results.length).isEqualTo(2);
+        assertThat(results[0]).isEqualTo(transportOffer1);
+        assertThat(results[1]).isEqualTo(transportOffer2);
+    }
+
+    @Test
+    void shouldFindByTransportDate() {
+        TransportOffer transportOffer1 = postOffer(aTransportOffer()
+                .transportDate(LocalDate.of(2022, 3, 21))
+                .build());
+        postOffer(aTransportOffer()
+                .transportDate(LocalDate.of(2022, 3, 22))
+                .build());
+
+        TransportOfferSearchCriteria searchCriteria = new TransportOfferSearchCriteria();
+        searchCriteria.setTransportDate(LocalDate.of(2022, 3, 21));
+        TransportOffer[] results = searchOffers(searchCriteria);
+
+        assertThat(results.length).isEqualTo(1);
+        assertThat(results[0]).isEqualTo(transportOffer1);
+    }
+
+    @Test
+    void shouldReturnPageOfData() {
+        postOffer(aTransportOffer().title("a").build());
+        postOffer(aTransportOffer().title("b").build());
+        postOffer(aTransportOffer().title("c").build());
+        postOffer(aTransportOffer().title("d").build());
+        postOffer(aTransportOffer().title("e").build());
+        postOffer(aTransportOffer().title("f").build());
+
+        PageRequest page = PageRequest.of(1, 2);
+        PageableResponse<TransportOffer> results = searchOffers(page);
+
+        assertThat(results.totalElements).isEqualTo(6);
+        assertThat(results.content)
+                .hasSize(2)
+                .extracting(r -> r.title)
+                .containsExactly("c", "d");
+    }
+
+    @Test
+    void shouldSortResults() {
+        postOffer(aTransportOffer().title("a").build());
+        postOffer(aTransportOffer().title("b").build());
+        postOffer(aTransportOffer().title("c").build());
```

Comment:
I wonder how it will work with polish/ukrainian diacritic letters. Maybe add here `ą` and check?

Reply:
good idea! I will add that